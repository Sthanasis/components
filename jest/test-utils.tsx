import { render as tlRender, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from 'src/assets/theme';

const render = (ui: ReactNode, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return {
    user: userEvent.setup(),
    ...tlRender(<ThemeProvider theme={defaultTheme}>{ui}</ThemeProvider>),
  };
};

export { render, fireEvent };
