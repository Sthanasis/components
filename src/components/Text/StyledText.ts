import { ThemeType } from 'src/utilities/theme';
import styled from 'styled-components';
import type { TextTagType, TextVariantType } from './Text';

interface IStyledTextProps {
  theme: ThemeType;
  variant: TextVariantType;
}

const getTextColor = ({ theme, variant }: IStyledTextProps) => {
  if (variant === 'primary') return theme.palette.primary.main;
  if (variant === 'secondary') return theme.palette.secondary.main;
  return 'inherit';
};

const StyledSpan = styled.span`
  color: ${getTextColor};
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-size: 15px;
`;
const StyledParagraph = styled.p`
  color: ${getTextColor};
  font-size: 15px;
`;
const StyledH1 = styled.h1`
  color: ${getTextColor};
  font-weight: inherit;
`;
const StyledH2 = styled.h2`
  color: ${getTextColor};
  font-weight: inherit;
`;
const StyledH3 = styled.h3`
  color: ${getTextColor};
  font-weight: inherit;
`;
const StyledH4 = styled.h4`
  color: ${getTextColor};
  font-weight: inherit;
`;
const StyledH5 = styled.h5`
  color: ${getTextColor};
  font-weight: inherit;
`;
const StyledH6 = styled.h6`
  color: ${getTextColor};
  font-weight: inherit;
`;
export const getStyledText = (tag: TextTagType) => {
  const componentMap = {
    span: StyledSpan,
    p: StyledParagraph,
    h1: StyledH1,
    h2: StyledH2,
    h3: StyledH3,
    h4: StyledH4,
    h5: StyledH5,
    h6: StyledH6,
  };
  return componentMap[tag];
};
