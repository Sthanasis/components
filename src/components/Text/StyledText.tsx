import styled from 'styled-components';
import type { TextTagType } from './Text';

const StyledSpan = styled.span`
  font-size: 18px;
`;
const StyledParagraph = styled.p`
  font-size: 15px;
`;
const StyledH1 = styled.h1`
  font-weight: inherit;
`;
const StyledH2 = styled.h2`
  font-weight: inherit;
`;
const StyledH3 = styled.h3`
  font-weight: inherit;
`;
const StyledH4 = styled.h4`
  font-weight: inherit;
`;
const StyledH5 = styled.h5`
  font-weight: inherit;
`;
const StyledH6 = styled.h6`
  font-weight: inherit;
`;
export const getStyledText = (tag: TextTagType) => {
  const componentMap = {
    span: StyledSpan,
    p: StyledParagraph,
    h1: StyledH1,
    h2: StyledH2,
    h3: StyledH3,
    h4: StyledH4,
    h5: StyledH5,
    h6: StyledH6,
  };
  return componentMap[tag];
};
