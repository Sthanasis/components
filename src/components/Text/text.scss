.text {
  &.base-text {
    font-size: 18px;
  }
  &.small-text {
    font-size: 15px;
  }
  &.header-text {
    font-weight: inherit;
  }
}
