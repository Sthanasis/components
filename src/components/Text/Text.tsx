import { IBaseProps } from 'src/types';
import { getStyledText } from './StyledText';

export type TextTagType =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'h5'
  | 'h6'
  | 'p'
  | 'span';

export type TextVariantType = 'primary' | 'secondary' | 'default';
export interface ITextProps extends IBaseProps {
  children: React.ReactNode;
  tag?: TextTagType;
  variant?: TextVariantType;
}

const Text = ({
  children,
  tag = 'span',
  className,
  variant = 'default',
  style,
}: ITextProps): JSX.Element => {
  const Component = getStyledText(tag);
  return (
    <Component variant={variant} className={className} style={style}>
      {children}
    </Component>
  );
};

export default Text;
