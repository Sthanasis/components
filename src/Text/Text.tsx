import { IBaseProps } from '../../../types/props';
import './text.scss';

interface ITextProps extends IBaseProps {
  children: React.ReactNode;
  variant?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'span';
}

const Text = ({
  children,
  variant = 'span',
  className,
  style,
}: ITextProps): JSX.Element => {
  const Component = variant;
  const classList = ['text'];
  if (variant === 'span') {
    classList.push('base-text');
  } else if (variant === 'p') {
    classList.push('small-text');
  } else {
    classList.push('header-text');
  }

  if (className) {
    className
      .trim()
      .split(' ')
      .forEach((c) => classList.push(c));
  }

  return (
    <Component className={classList.join(' ')} style={style}>
      {children}
    </Component>
  );
};

export default Text;
