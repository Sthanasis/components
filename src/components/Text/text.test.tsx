import { render } from 'test-utils';
import Text from '../Text';

describe('Text component', () => {
  it('renders the inside text', () => {
    const { getByText } = render(<Text>TEST</Text>);
    expect(getByText('TEST')).toBeInTheDocument();
  });
  it('renders an element based on the variant prop', () => {
    const { container } = render(<Text tag="h1">TEST</Text>);
    expect(container.firstChild?.nodeName.toLowerCase()).toBe('h1');
  });
  it('renders passes the className prop to element classes', () => {
    const { getByText } = render(<Text className="test test1">TEST</Text>);
    const el = getByText('TEST');
    const includesClasses =
      el.classList.contains('test') && el.classList.contains('test1');
    expect(includesClasses).toBe(true);
  });
});
