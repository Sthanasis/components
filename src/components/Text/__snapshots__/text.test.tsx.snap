// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component renders the inside text 1`] = `
<span
  class="sc-bcXHqe jyJnfJ"
>
  TEST
</span>
`;
